### ISSUE TITLE
Python packages versions

### ISSUE THREAD

**Issue by ChrisLinnNuvisan** [OBrink/DECIMER.ai#70]
Hi,

I updated our Decimer.ai installation today as I saw that there were lots of commits since the installation.

Thank you for your effort! Great tool.

However, I was not able to run it out of the box. Again I had issues with versions not matching the expected versions.

This is my Dockerfile pip install section now.
I needed to raise the tensorflow version to 2.12.0 and needed to add tensorflow-addons. Numpy needed a newer version, as well as keras. Protobuf was also now needed in the newest version.

Pip was complaining about needed but not installed packages: pyyaml, attrs, ipyplot, so I added them, don't know if they are really needed.

As far as I can tell, everything seems to be running. I can upload files and get results back. Could you please check what versions are currently really needed? As you're not telling pip which versions of DECIMER-Image_Transformer, decimer-segmentation, Smiles-TO-iUpac-Translator and DECIMER-Image-Classifier to install, I think this breaks things, when those packages change their needs. You should refer to specific versions to keep it reproducable.

```
RUN pip3 install pillow==9.4.0
RUN pip3 install --upgrade setuptools pip
RUN pip3 install opencv-python==4.7.0.68
RUN pip3 install matplotlib==3.4.3
RUN pip3 install scikit-image==0.19.3
RUN pip3 install imantics==0.1.12
RUN pip3 install pdf2image==1.16.2
RUN pip3 install IPython==8.10.0
RUN pip3 install pystow==0.4.9
RUN pip3 install unicodedata2==15.0.0
RUN pip3 install efficientnet==1.1.1
RUN pip3 install selfies==2.1.0
RUN pip3 install tensorflow==2.12.0
RUN pip3 install tensorflow-addons
RUN pip3 install keras==2.12.0
RUN pip3 install rdkit-pypi==2022.09.4
RUN pip3 install jpype1==1.4.1
RUN pip3 install protobuf
RUN pip3 install numpy==1.22
RUN pip3 install pyyaml
RUN pip3 install attrs
RUN pip3 install ipyplot
RUN pip3 install pillow-heif==0.11.1
RUN pip3 install --no-deps git+https://github.com/Kohulan/DECIMER-Image_Transformer
RUN pip3 install --no-deps decimer-segmentation
RUN pip3 install --no-deps git+https://github.com/Kohulan/Smiles-TO-iUpac-Translator/
RUN pip3 install --no-deps git+https://github.com/Iagea/DECIMER-Image-Classifier
```

Thank you very much in advance to have a look at this issue. Keep up the good work.

Best regards,
Chris

**Comment by OBrink** [comment]
Hey Chris,

Thank you for bringing this to our attention!

I think we broke it today with this PR that changed the behaviour of DECIMER Image Transformer: https://github.com/Kohulan/DECIMER-Image_Transformer/pull/67, and I forgot to make the necessary changes here. You are completely right, that this should not be the case at all, and all the versions of DECIMER Segmentation, STOUT, DECIMER Image Classifier and DECIMER Image Transformer should be specified. Tomorrow, I will update the dependencies and specify **all** versions so that everything is reproducible at any point in time.

I will tag this issue and drop a message here when it's done. Thanks a lot!

Otto

**Comment by OBrink** [comment]
Hey @ChrisLinnNuvisan,

All versions are specified now, and I have tested that it works out of the box again. I have specified DECIMER Image Transformer v2.3.1; so, Tensorflow 2.12 is not necessary because this does not include the changes we made yesterday (which have no influence on the predictions).

Again, thank you for opening the issue here! Let us know if you run into any further problems.

Have a nice day!
Otto
